recipes/armadillo/all/conanfile.py

```diff
@@ -0,0 +1,414 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class ArmadilloConan(ConanFile):
+    name = "armadillo"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://arma.sourceforge.net"
+    description = "Armadillo is a high quality C++ library for linear algebra and scientific computing, aiming towards a good balance between speed and ease of use."
+    topics = (
+        "linear algebra",
+        "scientific computing",
+        "matrix",
+        "vector",
+        "math",
+        "blas",
+        "lapack",
+        "mkl",
+        "hdf5",
+    )
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_blas": [
+            False,
+            "openblas",
+            "system_openblas",
+            "system_blas",
+            "system_mkl",
+            "system_flexiblas",
+            "framework_accelerate",
+        ],
+        "use_lapack": [
+            False,
+            "openblas",
+            "system_openblas",
+            "system_lapack",
+            "system_mkl",
+            "system_atlas",
+            "framework_accelerate",
+        ],
+        "use_hdf5": [False, "hdf5", "system_hdf5"],
+        "use_superlu": [False, "system_superlu"],
```

Comment:
L't's try to use only Conan dependencies, `"use_hdf5": [True, False]`. Why? Any error when using system libs will be not reproducible. Most cases we have that hard to solve are related to environment. Broken packages we can fix, but reproduce a custom environment is hard.

Reply:
Happy to apply this wherever there's a conan substitute, i.e. for openblas and hdf5. I understand your point, but I strongly suggest we keep the other system options (system_mkl, system_lapack, system_blas, system_flexiblas, system_atlas, system_superlu, system_arpack) until we're able to provide substitutes within the conan ecosystem. 

Armadillo loses a significant amount of utility if it can't be used with lapack specifically, and a lot of its value comes from being able to change its efficiency for particular purposes by changing the backend BLAS/LAPACK implementations. Most of these system deps provide a BLAS/LAPACK interface, and can't be included in the conan ecosystem until a fortran compiler is made available, either as a recipe or on the CI runners.

None of these system dependencies are enabled by default, so they require a user to make an effort to enable them. I think it's far more valuable to provide the full functionality as an option here - this gives the user the choice to make with respect to features or reproducibility.